src/Debugger-Oups/OupsDebuggerSystem.class.st

```diff
@@ -83,21 +82,13 @@ OupsDebuggerSystem >> logDebugRequest: aDebugRequest withUIManager: aCommandLine
 
 { #category : 'open debugger' }
 OupsDebuggerSystem >> openDebuggerOnRequest: aDebugRequest [
-
 	"Prepare the opening of a debugger, and delegate the debugger selection and opening to a strategy"
 
 	<debuggerCompleteToSender>
-	| debuggerOpeningStrategy |
 	aDebugRequest debugSession logStackToFileIfNeeded.
 	self performPreDebugActionsIn: aDebugRequest.
-	self spawnNewUIProcessIfNecessary: aDebugRequest.
 	self ensureExceptionIn: aDebugRequest debugSession.
-
-	"Schedule the debugger opening in a deferred UI message to address redraw problems after opening a debugger e.g. from the testrunner."
-	debuggerOpeningStrategy := self debuggerSelectionStrategy.
-	self defaultUIManager defer: [
-		debuggerOpeningStrategy openDebuggerForSession:
-			aDebugRequest debugSession ].
```

Comment:
Could you comment somewhere in the PR where is the code replacing this functionality? Is it in Spec? Somewhere else? is it WIP?

Reply:
Sorry, it is there https://github.com/pharo-spec/NewTools/pull/667
The spec debugger asks its application for the used backend, which in turn gives back a debug session specialized for that backend (Morphic or Gtk).
The specialized debug session is now handling the removed code that you pointed out.

It also means that if I want to create a new debugger, and if it should work with Morphic, I also have to ask the backend and the specialized session what to do.

Reply:
In other words: Morphic affairs are now Morphic affairs only.